unit_tests/bigquery/test_table.py

```diff
@@ -1449,6 +1449,12 @@ class TextModeFile(object):
         with self.assertRaises(ValueError):
             table.upload_from_file(file_obj, 'CSV', size=1234)
 
+    def test_upload_from_file_binary_mode_success(self):
+        try:
+            self._upload_from_file_helper(input_file_mode='r+b')
+        except ValueError:
+            self.fail("Unexpected failure when uploading binary mode file")
```

Comment:
@sebbacon If the test errors, we fix it, just like a failure:  it shouldn't need special handling here.